iron-overlay-behavior.html

```diff
@@ -245,24 +245,28 @@
       // Used for wrapping the focus on TAB / Shift+TAB.
       this.__firstFocusableNode = this.__lastFocusableNode = null;
       // Used for requestAnimationFrame when opened changes.
-      this.__openChangedAsync = null;
+      this.__openedChangedRAF = null;
       // Used for requestAnimationFrame when iron-resize is fired.
-      this.__onIronResizeAsync = null;
+      this.__ironResizeRAF = null;
+      // Focused node before overlay gets opened. Can be restored on close.
+      this.__restoreFocusNode = null;
       this._ensureSetup();
     },
 
     attached: function() {
       // Call _openedChanged here so that position can be computed correctly.
       if (this.opened) {
-        this._openedChanged();
+        this._openedChanged(this.opened);
       }
       this._observer = Polymer.dom(this).observeNodes(this._onNodesChange);
     },
 
     detached: function() {
       Polymer.dom(this).unobserveNodes(this._observer);
       this._observer = null;
-      this.opened = false;
+      this.__openedChangedRAF && window.cancelAnimationFrame(this.__openedChangedRAF);
```

Comment:
It won't remove it, but I can change it back to an `if` statement if you think is more readable ;)
https://closure-compiler.appspot.com/home#code%3D%252F%252F%2520%253D%253DClosureCompiler%253D%253D%250A%252F%252F%2520%2540compilation_level%2520ADVANCED_OPTIMIZATIONS%250A%252F%252F%2520%2540output_file_name%2520default.js%250A%252F%252F%2520%253D%253D%252FClosureCompiler%253D%253D%250A%250A%252F%252F%2520ADD%2520YOUR%2520CODE%2520HERE%250Afunction%2520updateObj(obj%252C%2520name)%2520%257B%250A%2520%2520obj.name%2520%2526%2526%2520alert('old%2520name%2520'%2520%252B%2520obj.name)%253B%250A%2520%2520obj.name%2520%253D%2520name%253B%250A%2520%2520obj.name%2520%2526%2526%2520alert('new%2520name%2520'%2520%252B%2520obj.name)%253B%250A%257D%250AupdateObj(%257B%257D%252C'New%2520user')%253B%250A%250A